spec/controllers/projects/contributions_controller_spec.rb

```diff
@@ -110,6 +110,25 @@
     end
   end
 
+  describe 'unavailable contribution projects' do
+    let(:project) { create(:project, available_for_contribution: false) }
+    let(:reward)  { create(:reward, project: project) }
+
+    context 'new' do
```

Comment:
Looking the code, the hooks aren't related to the actions, they are related to the state of the project. For this reason, the hook being tested here can happen on the `post` action too.